lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx

```diff
@@ -199,6 +200,12 @@ export default function AssignmentActivity() {
     });
   }, [students]);
 
+  const statusForUser = useCallback(
+    (h_userid: string) =>
+      lastSync.data?.grades.find(s => s.h_userid === h_userid)?.status,
+    [lastSync.data?.grades],
```

Comment:
There is an n^2 cost to the grade lookup (n student rows, n lookups per row). For sufficiently long lists of students it would be worth creating a memo-d map of student ID to sync result.

Reply:
Good point